From 59691a3f2a1696c160b020867be54b800d860206 Mon Sep 17 00:00:00 2001
From: Piotr Betlej <piotr.betlej@googlemail.com>
Date: Mon, 27 Oct 2025 16:14:15 +0100
Subject: [PATCH] Pymavlink as yocto dependency

Message_definitions installed into include directory. Use native
python3-pymavlink instead of vendored pip dependencies.

Upstream-Status: Inappropriate [mavsdk yocto integration]

---
 CMakeLists.txt | 31 ++++++++++++++-----------------
 1 file changed, 14 insertions(+), 17 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8aadc9af..10374fb2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,17 +4,6 @@ project(mavlink)
 
 find_package(Python COMPONENTS Interpreter REQUIRED)
 
-# We automatically install the pip dependencies locally below.
-# Therefore, we check whether pip is available here.
-execute_process(
-    COMMAND ${Python_EXECUTABLE} -m pip -V
-    RESULT_VARIABLE EXIT_CODE
-    OUTPUT_QUIET
-)
-if (NOT ${EXIT_CODE} EQUAL 0)
-    message(FATAL_ERROR "Python pip not found, pip is required")
-endif()
-
 if (NOT MAVLINK_DIALECT)
     set(MAVLINK_DIALECT common)
 endif()
@@ -28,16 +17,14 @@ message(STATUS "MAVLink version: ${MAVLINK_VERSION}")
 set(EXAMPLE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/${MAVLINK_DIALECT}/mavlink.h)
 
 add_custom_command(OUTPUT ${EXAMPLE_HEADER}
-    COMMAND ${Python_EXECUTABLE}
-        -m pip install -r pymavlink/requirements.txt --upgrade -t ${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/
-        COMMAND ${CMAKE_COMMAND} -E env "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/" ${Python_EXECUTABLE}
+    COMMAND ${CMAKE_COMMAND} -E env "PYTHONPATH=${PYTHON_NATIVE_SITEPACKAGES}" ${Python_EXECUTABLE}
         -m pymavlink.tools.mavgen
         --lang=C
         --wire-protocol=${MAVLINK_VERSION}
         --output ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/
-        message_definitions/v1.0/${MAVLINK_DIALECT}.xml
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    DEPENDS message_definitions/v1.0/${MAVLINK_DIALECT}.xml
+        ${CMAKE_CURRENT_SOURCE_DIR}/message_definitions/v1.0/${MAVLINK_DIALECT}.xml
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/message_definitions/v1.0/${MAVLINK_DIALECT}.xml
     COMMENT "Generating C headers")
 
 # Unfortunately, the dependencies don't work for INTERFACE libraries.
@@ -68,6 +55,14 @@ install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/include/mavlink"
     FILES_MATCHING PATTERN "*.h"
 )
 
+install(FILES
+    "${CMAKE_CURRENT_SOURCE_DIR}/message_definitions/v1.0/minimal.xml"
+    "${CMAKE_CURRENT_SOURCE_DIR}/message_definitions/v1.0/standard.xml"
+    "${CMAKE_CURRENT_SOURCE_DIR}/message_definitions/v1.0/common.xml"
+    "${CMAKE_CURRENT_SOURCE_DIR}/message_definitions/v1.0/ardupilotmega.xml"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/mavlink/message_definitions/v1.0"
+)
+
 install(EXPORT MAVLinkTargets
     FILE MAVLinkTargets.cmake
     NAMESPACE MAVLink::
